### PR TITLE
Fix SDK browser compatibility

### PR DESCRIPTION
**Pull request by pawelgrzybek**
## What does this implement/fix? Explain your changes.

It removes references to APIs incompatible with the browser. 

## Any other comments?

To make it work it was necessary to remove a `name` field on the child logger. As discussed with @israelboudoux, it is a little impact, and in case of `error` or `fatal` log, we will have the same level of info provided in the `stack` field.

Also, as part of this PR I also removed a cached files to ignored siles inside `./dist` directory. This show be present only on local and pipeline env. Not on the repo.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -8,8 +8,11 @@
     "nightfall_3",
     "nf3"
   ],
-  "main": "libs/user/index.js",
-  "module": "libs/user/index.js",
+  "main": "./libs/user/index.js",
+  "exports": {
+    "import": "./esm/user/index.js",
+    "require": "./cjs/user/index.js"
+  },
```
Comment (imagobea) [inline package.json:15]: Nice one

File: libs/user/user.ts [hunk 2]
```diff
@@ -1,4 +1,4 @@
-import path from "path";
+import path from "node:path";
```
Comment (imagobea) [inline libs/user/user.ts:1]: Delete `path` import, please double-check but pretty sure that after the `logger` change we no longer need it

File: libs/client/client.ts [hunk 3]
```diff
@@ -1,16 +1,11 @@
 import axios from "axios";
 import type { Commitment } from "../nightfall/types";
-import path from "path";
-import { parentLogger } from "../utils";
+import { logger } from "../utils";
 import type { NightfallZkpKeys } from "../nightfall/types";
 import type { RecipientNightfallData } from "libs/transactions/types";
-import { NightfallSdkError } from "../utils/error";
+import { NightfallSdkError } from "../utils";
```
Comment (imagobea) [inline libs/client/client.ts:6]:
```suggestion
import { logger, NightfallSdkError } from "../utils";
```

Or are you keeping them separate for readability (?)

On the other hand, if you accept the suggestion, please update the other `NightfallSdkError` imports :pray:

File: libs/utils/index.ts [hunk 4]
```diff
@@ -1,3 +1,4 @@
-import parentLogger from "./logger";
+import logger from "./logger";
+import { NightfallSdkError } from "./error.js";
```
Comment (imagobea) [inline libs/utils/index.ts:2]: :female_detective: 

```suggestion
import { NightfallSdkError } from "./error";
```

**Review by israelboudoux — APPROVED** [review]
LGTM

**Review by imagobea — CHANGES_REQUESTED** [review]
Hey @pawelgrzybek 
Thanks for the clean up in the SDK!

The `dist` was published by the semantic-release bot. We all agree that it's not needed on the repo. Perhaps we are missing something on the semantic-release config.

On the other hand, I couldn't run the example scripts - see error below. Other than that I just left a few comments.

Let me know if you have questions or need any help.
Cheers! :taco: 

```
bea@bea-pc:~/Dev/Polygon/nightfall-sdk$ npm run eg:ganache:balances

> nightfall-sdk@1.0.0 eg:ganache:balances
> ts-node -r dotenv/config examples/scripts/balances.ts dotenv_config_path=./examples/scripts/.env.ganache

(node:1352209) Warning: To load an ES module, set "type": "module" in the package.json or use the .mjs extension.
(Use `node --trace-warnings ...` to show where the warning was created)
/home/bea/Dev/Polygon/nightfall-sdk/examples/scripts/balances.ts:1
import { UserFactory } from "../../libs/user";
^^^^^^

SyntaxError: Cannot use import statement outside a module
    at wrapSafe (internal/modules/cjs/loader.js:979:16)
    at Module._compile (internal/modules/cjs/loader.js:1027:27)
    at Module.m._compile (/home/bea/Dev/Polygon/nightfall-sdk/node_modules/ts-node/src/index.ts:1618:23)
    at Module._extensions..js (internal/modules/cjs/loader.js:1092:10)
    at Object.require.extensions.<computed> [as .ts] (/home/bea/Dev/Polygon/nightfall-sdk/node_modules/ts-node/src/index.ts:1621:12)
    at Module.load (internal/modules/cjs/loader.js:928:32)
    at Function.Module._load (internal/modules/cjs/loader.js:769:14)
    at Function.executeUserEntryPoint [as runMain] (internal/modules/run_main.js:72:12)
    at phase4 (/home/bea/Dev/Polygon/nightfall-sdk/node_modules/ts-node/src/bin.ts:649:14)
    at bootstrap (/home/bea/Dev/Polygon/nightfall-sdk/node_modules/ts-node/src/bin.ts:95:10)

```

File: libs/client/client.ts (hunk 3 above)
Comment (pawelgrzybek) [reply libs/client/client.ts:6]: Very good suggestion, I will correct it across the project. Thanks 🙏

File: libs/utils/index.ts (hunk 4 above)
Comment (pawelgrzybek) [reply libs/utils/index.ts:2]: 🙏

**Review by imagobea — APPROVED** [review]
Hi @pawelgrzybek 
Thank you for the changes

It is amazing how a few small changes here and there can make such a big difference. Thank you for your magic. This week is a bit packed, but perhaps next week we can use some SDK time and learn from this (?) :pray: :nerd_face: 

Many thanks again!

**Review by luizoamorim — APPROVED** [review]
Sounds very good Pawel. I see that you changed process.env by globalThis. It makes total sense.

**Review by israelboudoux — APPROVED** [review]

**Pull request merged**